File: PgsKanban_Frontend/src/app/models/responses/board/userboard.ts
import { BoardMiniature } from './boardMiniature';
export interface UserBoard {
    userId: string;
    boardId: number;
    isOwner: boolean;
    isFavorite: boolean;
    board: BoardMiniature;
    lastTimeVisited: Date;
    lastTimeSetFavorite: Date;
}

File: PgsKanban_Backend/PgsKanban.DataAccess/Models/UserBoard.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using PgsKanban.DataAccess.Interfaces;

namespace PgsKanban.DataAccess.Models
{
    public class UserBoard : IDbEntity
    {
        [Column(Order = 0), Key, ForeignKey("AspNetUsers")]
        public string UserId { get; set; }
        [ForeignKey("UserId")]
        public virtual User User { get; set; }
        [Column(Order = 1), Key, ForeignKey("Boards")]
        public int BoardId { get; set; }
        [ForeignKey("BoardId")]
        public virtual Board Board { get; set; }
        public bool IsFavorite { get; set; }
        public DateTime LastTimeVisited { get; set; }
        public DateTime LastTimeSetFavorite { get; set; }
        public bool IsDeleted { get; set; }
    }
}
